eng/scripts/Find-InvalidSlug.ps1

```diff
@@ -0,0 +1,451 @@
+[CmdletBinding()]
+param (
+    [Parameter(Position=0, Mandatory=$true, ValueFromPipeline=$true)]
+    [string[]] $Path,
+
+    [Parameter()]
+    [switch] $IncludeParents
+)
+
+process {
+    foreach ($p in $Path) {
+        $file = Get-Item $p
+        if ($file.Extension -ne '.md') {
+            Write-Verbose "'$($file.FullName)' does not appear to be a valid markdown file"
+            continue
+        }
+
+        [string[]] $content = $file | Get-Content
+        if (!$content[0].StartsWith('---')) {
+            Write-Verbose "No markdown frontmatter found in '$($file.FullName)'"
+            continue
+        }
+
+        $i = 1
+        $products = @()
+        do {
+            [string] $line = $content[$i++]
+            if ($line.StartsWith('---')) {
+                break
+            }
+
+            if ($line -match '^\s*products:') {
+                $found = $true
+            }
+            elseif ($found -and $line -match '^\s*-\s+([\w-]+)') {
+                $products += $matches[1]
+            }
+            elseif (![string]::IsNullOrWhiteSpace($line)) {
+                $found = $false
+            }
+        } while ($i -lt $content.Length)
+
+        $invalidProducts = @()
+        foreach ($product in $products) {
+            if ($slugs -notcontains $product) {
+                $invalidProducts += $product
+                Write-Error "File '$($file.FullName)' contains invalid slug: $product"
+            }
+        }
+
+        if ($invalidProducts) {
+            $file | Add-Member -Type NoteProperty -Name 'InvalidSlugs' -Value $invalidProducts -PassThru
+        }
```

Comment:
Since you've changed `Write-Warning` to `Write-Error`, remove this block. I wrote the original script to be more PowerShelly, i.e. not doing a monolithic thing, but instead returning results on which you could act on further.

Reply:
I fixed this in my updates (see other comment).